dcicutils/misc_utils.py

```diff
@@ -1560,6 +1567,110 @@ def singleton_class(self):
         return self._singleton_class
 
 
+class classproperty(object):
+    """
+    This decorator is like 'classproperty', but the function is run only on first use, not every time, and then cached.
+
+    Example:
+
+        import time
+        class Clock:
+            @classproperty
+            def sample():
+                return time.time()
+
+        # Different results each time, just like an instance method, but without having to instantiate the class.
+        Clock.sample
+        1665600812.008385
+        Clock.sample
+        1665600812.760394
+    """
+
+    def __init__(self, getter):
+        self.getter = getter
+
+    def __get__(self, instance, instance_class):
+        ignored(instance)
+        return self.getter(instance_class)
+
+
+class classproperty_cached(object):
```

Comment:
You might consider providing a "reset" method that will "delete" the cached class property (so it is recomputed). I can see how this is useful but I also think under some situations you may want to trigger a discard of the cached value.